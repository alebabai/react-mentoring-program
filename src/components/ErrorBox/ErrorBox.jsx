import React from 'react'

export const ErrorBox = ({ message, description }) => (
    <div className="error-box">
        <h1 className="error-box-message">{message}</h1>
        <p className="error-box-description">{description}</p>
    </div>
)

export default ErrorBox
