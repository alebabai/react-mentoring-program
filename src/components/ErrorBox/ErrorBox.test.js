import React from 'react'

import ErrorBox from './ErrorBox'

import renderer from 'react-test-renderer';

const props = {
    message: 'Something wrong',
    description: 'An error happened'
}

describe('Should match snapshot', () => {
    test('with default props', () => {
        const component = renderer.create(
            <ErrorBox />
        )
        expect(component.toJSON()).toMatchSnapshot()
    })

    test('with all props', () => {
        const component = renderer.create(
            <ErrorBox {...props} />
        )
        expect(component.toJSON()).toMatchSnapshot()
    })
})
