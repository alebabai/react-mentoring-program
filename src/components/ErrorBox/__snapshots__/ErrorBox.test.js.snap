// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot with all props 1`] = `
<div
  className="error-box"
>
  <h1
    className="error-box-message"
  >
    Something wrong
  </h1>
  <p
    className="error-box-description"
  >
    An error happened
  </p>
</div>
`;

exports[`Should match snapshot with default props 1`] = `
<div
  className="error-box"
>
  <h1
    className="error-box-message"
  />
  <p
    className="error-box-description"
  />
</div>
`;
